### PR TITLE
DiskSelector Implmentation 

### PR DESCRIPTION
**Pull request by ravipwaghmare**
Added Disk Selector Implmentation.
Added Test Cases for Disk Selector.

This PR is related to the below proposal and issue:
https://github.com/metal3-io/metal3-docs/pull/152
https://github.com/metal3-io/hardware-classification-controller/issues/56

@dhellmann @maelk @digambar15 @russellb @zaneb 
Kindly review.

### CODE REVIEW

**Comment by fmuyassarov** [comment]
@ravipwaghmare Would you mind to add a reference in the PR description to the proposal of DiskSelector?

**Comment by Ashughorla** [comment]
@fmuyassarov Proposal link and issue link updated in description.

File: classifier/disk.go [hunk 1]
```diff
@@ -72,3 +95,58 @@ func checkRangeCapacity(min, max, count bmh.Capacity) bool {
...
+}
+
+// validatePattern finds out the disk with the pattern provided in hardware profile
+func validatePattern(HCTL string) string {
+
+	if HCTL == "0:0:0:0" {
+		return "0:0:N:0"
+
+	} else {
+		res1 := strings.Split(HCTL, ":")
```
Comment (digambar15) [inline classifier/disk.go]: variable should have numbers, change it to re/resp

File: classifier/disk.go [hunk 2]
```diff
@@ -72,3 +95,58 @@ func checkRangeCapacity(min, max, count bmh.Capacity) bool {
...
+		for index, st := range res1 {
+			value, _ := strconv.Atoi(st)
+			if value > 0 {
+				res1[index] = "N"
+			}
+		}
+		return strings.Join(res1, ":")
```
Comment (digambar15) [inline classifier/disk.go]: change var name res1 to resp every other places.

**Review by digambar15 — CHANGES_REQUESTED** [review]
Few minor comments. Rest looks like.

File: classifier/disk.go (hunk 2 above)
Comment (ravipwaghmare) [reply classifier/disk.go]: DONE

File: classifier/disk.go (hunk 1 above)
Comment (ravipwaghmare) [reply classifier/disk.go]: Done

**Review by digambar15 — APPROVED** [review]
Looks good.

**Comment by digambar15** [comment]
Please rebase the PR.

**Comment by digambar15** [comment]
Please rebase the PR

**Comment by ravipwaghmare** [comment]
> Please rebase the PR
Rebased Done

**Comment by digambar15** [comment]
Thanks!

**Pull request merged**